src/track.c

```diff
@@ -992,18 +998,20 @@ static void event(tracker_channel_t *tracker_channel, event_t event)
  * \param sample_count      Sample count.
  * \param carrier_freq      Carrier frequency.
  * \param cn0               C/N0 estimate.
+ * \param code              Code identifier.
  */
 static void common_data_init(tracker_common_data_t *common_data,
-                             u32 sample_count, float carrier_freq, float cn0)
+                             u32 sample_count, float carrier_freq,
+                             float cn0, code_t code)
 {
   /* Initialize all fields to 0 */
   memset(common_data, 0, sizeof(tracker_common_data_t));
 
   common_data->TOW_ms = TOW_INVALID;
 
   /* Calculate code phase rate with carrier aiding. */
-  common_data->code_phase_rate = (1 + carrier_freq/GPS_L1_HZ) *
-                                    GPS_CA_CHIPPING_RATE;
+  common_data->code_phase_rate = (1 + carrier_freq / code_to_carr_freq(code)) *
+                                 GPS_CA_CHIPPING_RATE;
   common_data->carrier_freq = carrier_freq;
```

Comment:
I agree. That would be great. It confused me already many times. And in this particular place it is as confusing as it could be.